app/processing/process-return-file.js

```diff
@@ -0,0 +1,35 @@
+const { downloadFile, archiveFile, quarantineFile } = require('../storage')
+const { DAX } = require('../constants/containers')
+const { parseAcknowledgementFile } = require('./parse-acknowledgement-file')
+const { sendDemographicsFailureEvent } = require('../event')
+const { DEMOGRAPHICS_UPDATE_FAILED } = require('../constants/events')
+
+const processReturnFile = async (file) => {
+  try {
+    console.log(`Processing return file: ${file}`)
+    const xmlData = await downloadFile(file, DAX)
+    const output = await parseAcknowledgementFile(xmlData)
+    if (output.failedFRNs[0]) {
+      let error = 'Demographics updates for the following FRNs have failed to be processed by DAX: '
+      for (let i = 0; i < output.failedFRNs.length; i++) {
```

Comment:
could this be improved by using .map to reduce the amount of potential looping?

Reply:
good spot, amended in latest code push